docs/src/main/paradox/jdbc.md

```diff
@@ -1,8 +1,222 @@
-# Offset in relational DB with JDBC
+# Offset in a relational DB with JDBC
 
-Not implemented yet, see [issue #23](https://github.com/akka/akka-projection/issues/23).
+The @apidoc[JdbcProjection] has support for storing the offset in a relational database using JDBC.
+
+The source of the envelopes can be @ref:[events from Akka Persistence](eventsourced.md) or any other `SourceProvider`
+with supported @ref:[offset types](#offset-types).
+
+A @apidoc[JdbcHandler] receives a @apidoc[JdbcSession] instance and an envelope. The `JdbcSession` provides the means to access an open JDBC connection that can be used to process the envelope. The target database operations can be run in the same transaction as the storage of the offset, which means that @ref:[exactly-once](#exactly-once)
+processing semantics is supported. It also offers @ref:[at-least-once](#at-least-once) semantics.
+
+## Dependencies
+
+To use the JDBC module of Akka Projections add the following dependency in your project:
+
+@@dependency [sbt,Maven,Gradle] {
+  group=com.lightbend.akka
+  artifact=akka-projection-jdbc_$scala.binary.version$
+  version=$project.version$
+}
+
+Akka Projections require Akka $akka.version$ or later, see @ref:[Akka version](overview.md#akka-version).
+
+@@project-info{ projectId="akka-projection-jdbc" }
+
+
+### Transitive dependencies
+
+The table below shows `akka-projection-jdbc`'s direct dependencies, and the second tab shows all libraries it depends on transitively.
+
+@@dependencies{ projectId="akka-projection-jdbc" }
+
+## Defining a JdbcSession
+
+Before using 
+> TODO - explain how and why we need one. Also the connection / thread pool relation. 
+
+Scala
+:  @@snip [JdbcProjectionDocExample.scala](/examples/src/test/scala/docs/jdbc/JdbcProjectionDocExample.scala) { #jdbc-session-imports #jdbc-session }
+
+Java
+:  @@snip [JdbcProjectionDocExample.java](/examples/src/test/java/jdocs/jdbc/JdbcProjectionDocExample.java) { #todo }
+
+## Blocking JDBC Dispatcher
+
+JDBC APIs are blocking by design, therefore Akka Projection JDBC will use a dedicated dispatcher to run all JDBC calls. It's important to configure the dispatcher to have the same size as the connection poll. 
+
+Each time an envelope is delivered to the handler one thread and one database connection will be used. If your connection pool is smaller than the number of threads, the thread can potentially block while waiting for the connection pool to provide a connection. 
+
+Most applications will use database connections to read data, for instance to read a projected model upon user request. This means that other parts of the application will be competing for a connection. It's recommend to configure a connection pool dedicated to the projections and use a different one in other parts of the application.  
```

Comment:
Submitting this for earlier feedback. 

The same way we have a dedicated dispatcher, we should have a dedicated connection pool. We manage the dispatcher, but we can't managed the connection pools. That opens the door for potentially badly configured systems. 

We could later make the `JdbcSessionUtils.withSession` public and recommend users to use it each time they need to do some JDBC. 

Anyway, for now we can give that recommendation. WDYT?